### PR TITLE
openstack api support doc

### PR DESCRIPTION
**Pull request by yb01**
**What type of PR is this?**
 /kind documentation

**What this PR does / why we need it**:
new openstack api support doc

### CODE REVIEW

File: docs/design-proposals/apiserver/openstack-api-support.md [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+## New APIs in Arktos
+### New API route paths:
+The new routes for servers
+
+* 	{Arktos-service-url}/servers
+* 	{Arktos-service-url}/servers/{server-id}/detail
+
+The new route for server actions
+
+* {Arktos-service-url}/servers/{server-id}/action
```
Comment (Sindica) [inline docs/design-proposals/apiserver/openstack-api-support.md:38]: It is unclear that how different actions will be differentiated in URL and request body

File: docs/design-proposals/apiserver/openstack-api-support.md [hunk 2]
```diff
@@ -0,0 +1,615 @@
+---
+title: Support Openstack Compute APIs in Arktos
+authors:
+  - "@yb01"
+---
+
+# Support Openstack Compute APIs in Arktos
```
Comment (Sindica) [inline docs/design-proposals/apiserver/openstack-api-support.md:7]: Can you add a TOC at the beginning? This document is very long and layout is not clear.

File: docs/design-proposals/apiserver/openstack-api-support.md [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+## Background and context
+To boost Arktos VM adoption and to let Openstack users to easily explore Arktos VM support without a lot of changes from 
+the users’ applications, it is decided to add a new set of APIs in Arktos to match Openstack Server related operations. 
+Besides better user experiences, this new set ofAPIs will also help existing Openstack perf tests to be relatively easily
```
[the remark this answers is not in the document]
Comment (yb01) [reply docs/design-proposals/apiserver/openstack-api-support.md]: fixed.

File: docs/design-proposals/apiserver/openstack-api-support.md (hunk 2 above)
Comment (yb01) [reply docs/design-proposals/apiserver/openstack-api-support.md:7]: good idea. added TOC

File: docs/design-proposals/apiserver/openstack-api-support.md (hunk 1 above)
Comment (yb01) [reply docs/design-proposals/apiserver/openstack-api-support.md:64]: this is Openstack specific. the action details, action, action extra parameters etc, are all in the request body. the URL is followed what OS has.

Comment (yb01) [reply docs/design-proposals/apiserver/openstack-api-support.md:64]: this in the VM action section below.

**Review by Sindica — APPROVED** [review]
/lgtm

**Pull request merged**